test/src/test/java/org/axonframework/test/aggregate/ResultValidatorImplTest.java

```diff
@@ -72,9 +96,150 @@ void shouldConsiderExplicitEqualsBeforeCheckingFields() {
         validator.expectEvents(s2);
     }
 
+    @Test
+    void noDeadlineMatchingInTimeframeWithDeadlineInsideWindow() {
```

Comment:
I think there is a test case missing for the situation where there is a deadline scheduled inside the "deadline window", but that deadline does not match the matcher.